bukkit/src/main/java/com/github/games647/fastlogin/bukkit/hook/AuthMeHook.java

```diff
@@ -41,25 +58,32 @@ public void onSessionRestore(RestoreSessionEvent restoreSessionEvent) {
 
     @Override
     public boolean forceLogin(Player player) {
-        if (AuthMeApi.getInstance().isAuthenticated(player)) {
+        if (authmeAPI.isAuthenticated(player)) {
             plugin.getLog().warn(ALREADY_AUTHENTICATED, player);
             return false;
         }
 
         //skips registration and login
-        AuthMeApi.getInstance().forceLogin(player);
+        authmeAPI.forceLogin(player);
         return true;
     }
 
     @Override
     public boolean isRegistered(String playerName) {
-        return AuthMeApi.getInstance().isRegistered(playerName);
+        return authmeAPI.isRegistered(playerName);
     }
 
     @Override
+    //this automatically login the player too
     public boolean forceRegister(Player player, String password) {
-        //this automatically login the player too
-        AuthMeApi.getInstance().forceRegister(player, password);
-        return true;
+        //if we have the management - we can trigger register with IP limit checks
+        if (authmeManagement != null) {
+            authmeManagement.performRegister(RegistrationMethod.PASSWORD_REGISTRATION,
+                    ApiPasswordRegisterParams.of(player, password, true));
+        } else {
+            authmeAPI.forceRegister(player, password);
+        }
+
+        return authmeAPI.isRegistered(player.getName());
```

Comment:
Couldn't this cause an race-condition? If the calls are async, they would immediately return and therefore the check `authmeAPI.isRegistered` could return false even though the registration will be successful. 

This has to be tested with async enabled AuthMe.

Reply:
Yeah, I forgot to change that back to true - checking if isRegistered() sometimes works and sometime doesn't